vm/vm.go

```diff
@@ -157,35 +157,74 @@ func (mod) hex() []uint8 {
 	return []uint8{uint8(opcode.Mod)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (lt) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X < Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (lt) hex() []uint8 {
 	return []uint8{uint8(opcode.LT)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (gt) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X > Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (gt) hex() []uint8 {
 	return []uint8{uint8(opcode.GT)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (eq) Do(stack *stack, _ asmReader) error {
+	y := stack.pop()
+	x := stack.pop()
+
+	Y := int(y)
+	X := int(x)
+
+	if X == Y {
+		stack.push(item(1))
+	} else {
+		stack.push(item(0))
+	}
+
 	return nil
 }
 
 func (eq) hex() []uint8 {
 	return []uint8{uint8(opcode.EQ)}
 }
 
-// TODO: implement me w/ test cases :-)
 func (not) Do(stack *stack, _ asmReader) error {
+	x := stack.pop()
+
+	X := int(x)
+
+	// ~x = - x - 1
```

Comment:
Expression of `~x` in go is `^x`. how about `^X`?

Reply:
confirm thx 👍